asab/storage/service.py

```diff
@@ -159,6 +159,9 @@ def aes_encrypt(self, raw: bytes, iv: bytes = None) -> bytes:
 			else:
 				raise TypeError("Only 'bytes' objects can be encrypted")
 
+		# Append terminating character to separate padding from actual value
+		raw = raw + b"\xff"
```

Comment:
This smells. What do you do here exactly?

Reply:
it's in the description. i need to separate the actual value from the padding so im using a known arbitrary character (`\xff`).

### issue example
```python
secret = b"\xaa\xbb\x00"
u = StorageService.upsertor("coll", obj_id="abc")
u.set("secret", secret, encrypt=True)
await u.execute()

item = await StorageService.get("coll", obj_id="abc", decrypt=["secret"])
print(item["secret"])
>>> b"\xaa\xbb"
```

Reply:
Issues here are:

1) What if there is a proper `0xff` character at the end of the actual message - how you recognize it
2) I don't see stripping of it at the end

Proper ways of padding:

- you ensure that the actual message to encrypt is of the size required by the encryption so no padding is needed
- you have length field at the beginning of the message, so you now how long the message actually is

Reply:
Will this be a breaking change?

Reply:
i know what the issue is. but thanks for recommending the padding library, i did not know that.

it may be a breaking change. i will provide some back-compatibility.